nanome/api/schemas/structure_schemas.py

```diff
@@ -0,0 +1,177 @@
+import itertools
+from marshmallow import Schema, fields, post_load
+from nanome.api import structure
+from nanome.util import enums
+
+from .util_schemas import ColorField, EnumField, QuaternionField, Vector3Field
+
+
+class StructureSchema(Schema):
+    index = fields.Integer(default=-1)
+
+
+class AtomSchema(StructureSchema):
+    selected = fields.Boolean()
+    labeled = fields.Boolean()
+    atom_rendering = fields.Boolean()
+    surface_rendering = fields.Boolean()
+    exists = fields.Boolean()
+    is_het = fields.Boolean()
+    occupancy = fields.Boolean()
+    bfactor = fields.Boolean()
+    acceptor = fields.Boolean()
+    donor = fields.Boolean()
+    polar_hydrogen = fields.Boolean()
+    atom_mode = EnumField(enum=enums.AtomRenderingMode)
+    serial = fields.Integer()
+    current_conformer = fields.Integer(load_only=True)
+    conformer_count = fields.Integer(load_only=True)
+    positions = fields.List(Vector3Field())
+    label_text = fields.String()
+    atom_color = ColorField()
+    atom_scale = fields.Float()
+    surface_color = ColorField()
+    surface_opacity = fields.Float()
+    symbol = fields.Str()
+    name = fields.Str()
+    position = Vector3Field()
+    formal_charge = fields.Integer()
+    partial_charge = fields.Float()
+    vdw_radius = fields.Float(load_only=True)
+    alt_loc = fields.Str(max=1)
+
+    @post_load
+    def make_atom(self, data, **kwargs):
+        new_obj = structure.Atom()
+        for key in data:
+            try:
+                setattr(new_obj, key, data[key])
+            except AttributeError:
+                raise AttributeError('Could not set attribute {}'.format(key))
+        return new_obj
```

Comment:
I think for brevity, this should be refactored into a util method. I see this same for/try block repeated in all the schemas. something like:
```py
def make_atom(self, data, **kwargs):
    new_obj = init_obj(structure.Atom, data)
    # ... other schema-specific things
    return new_obj
```